### ISSUE TITLE
Addon-docs: 'No props found for this component' using Props table in .mdx file

### ISSUE THREAD

**Issue by anynines-johannchopin** [storybookjs/storybook#10656]
**Describe the bug**
Trying to document my React/TypeScript project using Storybook with `.mdx` files, I'm unable to make the Props table recognise my component's props. Using `<Props of={Button} />` I get this  message in the Props table field: 

> No props found for this component

The problem is that the component has declared props:

*Button.tsx:*
```typescript
...
export interface ButtonProps {
  type: ButtonType
  className?: string
  onClick?: () => void
}

export type ButtonType = 'primary' | 'dark' | 'darker' | 'default' | 'submit'

const Button: React.FunctionComponent<ButtonProps> = (props) => {
  const { children, onClick, type, className = '' } = props
  ...
}

export default Button
```
*button.stories.mdx:*
```markdown
import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks'

import Button from './Button'

<Meta title='Button' component={Button} />

# Badge

Let's define a story for our `Button` component:

<Preview>
    <Button 
      type='primary' 
      className='btn'
      onClick={() => alert('Click!')}
    >Click me</Button>
</Preview>

## Props 🔧

<Props of={Button} />
```

----

**To Reproduce**

1. Clone [this repo](https://github.com/anynines-johannchopin/addon-docs_Props-not-found-demo) that illustrate this bug
2. Install dependencies: `yarn install`
3. Run storybook: `yarn storybook`
4. Go to the [Button component doc](http://localhost:6006/?path=/docs/button--page) and see the empty **Props** section

----

**Expected behavior**
See in the **Props** section all the props of the `Button` component in a nice table

----

**System:**
```
  System:
    OS: macOS Mojave 10.14.6
    CPU: (8) x64 Intel(R) Core(TM) i7-4870HQ CPU @ 2.50GHz
  Binaries:
    Node: 12.16.1 - /usr/local/bin/node
    Yarn: 1.22.4 - /usr/local/bin/yarn
    npm: 6.13.4 - /usr/local/bin/npm
  Browsers:
    Chrome: 81.0.4044.129
    Safari: 13.1
  npmPackages:
    @storybook/addon-docs: ^5.3.17 => 5.3.18 
    @storybook/addons: ^5.3.17 => 5.3.18 
    @storybook/preset-create-react-app: ^2.1.1 => 2.1.1 
    @storybook/react: ^5.3.17 => 5.3.18 
```

**Comment by GasimGasimzada** [comment]
Did you install and configure `react-docgen-typescript-loader`?

**Comment by anynines-johannchopin** [comment]
I use in this project the `@storybook/preset-create-react-app` addon that handle TypeScript conf for me. Should I tried using `react-docgen-typescript-loader`?

**Comment by GasimGasimzada** [comment]
I don't think `@storybook/preset-create-react-app` supports `react-docgen-typescript-loader` because CRA does not have this configured and the addon uses CRA configuration. Can you try to install both `preset-create-react-app` and `preset-typescript` (not sure if this is a good idea) or install configure docgen manually using `webpackFinal` option.

**Comment by shilman** [comment]
@GasimGasimzada CRA preset supports `react-docgen` and this walkthrough should work:

https://gist.github.com/shilman/bc9cbedb2a7efb5ec6710337cbd20c0c

Older versions used `react-docgen-typescript-loader` and `1.5.2` was the most recent version that used it.

**Comment by shilman** [comment]
We've just released zero-config typescript support in 6.0-beta. I suspect this will solve most of your typescript props-related issues.

- [Release 6.0.0-beta.12+](https://github.com/storybookjs/storybook/releases/tag/v6.0.0-beta.12)
- [Migration instructions](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#zero-config-typescript)
- [Updated docs](https://github.com/storybookjs/storybook/blob/next/docs/src/pages/configurations/typescript-config/index.md)

Please try it out!
